interp/notation_ops.ml

```diff
@@ -882,19 +889,19 @@ let rec match_cases_pattern_binders metas acc pat1 pat2 =
 let glue_letin_with_decls = true
 
 let rec match_iterated_binders islambda decls = function
-  | GLambda (_,Name p,bk,t,GCases (_,LetPatternStyle,None,[(GVar(_,e),_)],[(_,_,[cp],b)]))
+  | GLambda (loc,Name p,bk,t,GCases (_,LetPatternStyle,None,[(GVar(_,e),_)],[(_,_,[cp],b)]))
       when islambda && Id.equal p e ->
-      match_iterated_binders islambda ((Inr cp,bk,None,t)::decls) b
-  | GLambda (_,na,bk,t,b) when islambda ->
-      match_iterated_binders islambda ((Inl na,bk,None,t)::decls) b
-  | GProd (_,Name p,bk,t,GCases (_,LetPatternStyle,None,[(GVar(_,e),_)],[(_,_,[cp],b)]))
+      match_iterated_binders islambda (GLocalPattern (loc,(cp,[](*dummy*)),p,bk,t)::decls) b
```

Comment:
This field should be a redundant precomputation of the set of variables declared by the patterns of the clause. It is then used by the compilation of pattern-matching.

In the current situation, the `GLocalPattern` expression is used  for printing, so it should not matter as it is, and this is what my comment `dummy` is supposed to indicate. However, it would matter in the case I found lowly probable that someone reenters pretyping with this expression.

Actually, I'm stupid. I just realized that we already have this field available, so we can do much better.

I pushed an extra fix. Since you already made the merge, could you just cherry-pick the extra commit and merge it to the one which introduced `dummy`. (Or, alternatively, give me a pointer to your merge and I'll do the cherry-pick and "fix" rebase). Thanks for having asked.